app/scripts/metamask-controller.js

```diff
@@ -4825,6 +4825,9 @@ export default class MetamaskController extends EventEmitter {
         log.error(err);
       }
     });
+
+    // Used to show wallet liveliness to the provider
+    this._notifyChainChange();
```

Comment:
note that this fires a `metamask_chainChanged` to all connections, not just the new one. Need to update the code to change this, but also it's starting to feel wrong using `metamask_chainChanged` as a liveliness check

Reply:
notify only new connection here https://github.com/MetaMask/metamask-extension/pull/24142/commits/f90b220061cd7252f2376b827ee9759dee236ae6